src/main/java/gregtech/common/covers/filter/SmartItemFilter.java

```diff
@@ -45,8 +53,10 @@ public Object matchItemStack(ItemStack itemStack) {
         if (cachedTransferRateValue == null) {
             ItemStack infinitelyBigStack = itemStack.copy();
             infinitelyBigStack.setCount(Integer.MAX_VALUE);
-            Recipe recipe = filteringMode.recipeMap.findRecipe(Long.MAX_VALUE, Collections.singletonList(infinitelyBigStack), Collections.emptyList(), Integer.MAX_VALUE);
-            if(recipe == null) {
+
+            Recipe recipe = filteringMode.recipeMap.findRecipe(Long.MAX_VALUE, Collections.singletonList(infinitelyBigStack), Collections.emptyList(), Integer.MAX_VALUE, matchingMode.matchingMode);
+
```

Comment:
Please don't leave empty lines in places that does not need them as separator of logic.
```suggestion
```